tests/4.5/taskloop/test_taskloop_private.c

```diff
@@ -0,0 +1,58 @@
+//---------------------------------------- test_taskloop_private.c -----------------------------------------//
+//
+// OpenMP API Version 4.5 August 2015
+//
+// This test checks the taskloop directive with the 'private' clause specified.
+// The 'taskloop' construct parallelizes loops with independent iterations by creating tasks. 
+// It allows for efficient parallel execution of loop iterations by distributing them among multiple threads. 
+// The 'private' clause ensures that each task should have private copies. 
+//----------------------------------------------------------------------------------------------------------//
+
+#include <stdio.h>
+#include <omp.h>
+#include "ompvv.h"
+
+#define NUM_THREADS 100
+#define NUM_TASKS 100
+
+int test_taskloop_private() {
+
+  int errors = 0;
+
+  int private_var = -2; 
+  int shared_var_sum = 0;	
+  int num_threads = 0;
+  
+   #pragma omp parallel num_threads(NUM_THREADS)
+   {
+      num_threads = omp_get_num_threads();
+      #pragma omp single
+      {
+	#pragma omp taskloop private(private_var) 
+        for(int i = 0; i < NUM_TASKS; i++) {
+	  private_var = omp_get_thread_num();
+          #pragma omp atomic
+	    shared_var_sum += private_var;
+	 }      
+      }	   
```

Comment:
```suggestion
      #pragma omp single
      {
	#pragma omp taskloop private(private_var) 
        for(int i = 0; i < NUM_TASKS; i++) {
          int sum = 0;
	  for(private_var = 0; private_var < 5; private_var++)
	    sum += private_var;
          #pragma omp atomic
	    shared_var_sum += sum;
	 }      
      }	   
```
This way we would use `private_var` to add `sum` to `10 (1+2+3+4)`. Each task would do the same. And below we can check that the each taskloop iteration did the increment using the `private_var`. Below we can test that the end value of `shared_var_sum` was `NUM_TASKS * 10`.

Reply:
@fel-cab: updated with your comments.